File: src/org/usfirst/frc/team4099/auto/modes/BaselineMode.java
package org.usfirst.frc.team4099.auto.modes;

// go to the baseline

import org.usfirst.frc.team4099.auto.AutoModeEndedException;
import org.usfirst.frc.team4099.auto.actions.ForwardAction;
import org.usfirst.frc.team4099.lib.util.AutonomousInitParameters;

/**
 * Created by plato2000 on 2/13/17.
 */
public class BaselineMode extends AutoModeBase {
    private final double initialForwardDistance;
    private double turningAngle;
    private boolean turnAround;

    public BaselineMode(AutonomousInitParameters initParameters, boolean turnAround) {
        this.turnAround = turnAround;
        this.initialForwardDistance = initParameters.getInitalForwardSeconds();
        this.turningAngle = initParameters.getTurnAngle().getDegrees();
    }

    @Override
    protected void routine() throws AutoModeEndedException {
        runAction(new ForwardAction(initialForwardDistance));
//        runAction(new TurnAction(Rotation2D.fromDegrees(60)));
    }


}

File: src/main/java/org/usfirst/frc/team4099/auto/modes/BaselineMode.kt
package org.usfirst.frc.team4099.auto.modes

// go to the baseline

import org.usfirst.frc.team4099.auto.AutoModeEndedException
import org.usfirst.frc.team4099.auto.actions.ForwardAction
import org.usfirst.frc.team4099.lib.util.AutonomousInitParameters

/**
 * Created by plato2000 on 2/13/17.
 */
class BaselineMode(initParameters: AutonomousInitParameters, private val turnAround: Boolean) : AutoModeBase() {
    private val initialForwardDistance: Double
    private val turningAngle: Double

    init {
        this.initialForwardDistance = initParameters.initalForwardSeconds
        this.turningAngle = initParameters.turnAngle.degrees
    }

    @Throws(AutoModeEndedException::class)
    override fun routine() {
        runAction(ForwardAction(initialForwardDistance))
        //        runAction(new TurnAction(Rotation2D.fromDegrees(60)));
    }


}
